test/widgets/subscription_list_test.dart

```diff
@@ -89,18 +89,71 @@ void main() {
     check(isUnpinnedHeaderInTree()).isTrue();
   });
 
-  testWidgets('subscription sort', (tester) async {
+  testWidgets('pinned subscriptions are sorted', (tester) async {
     await setupStreamListPage(tester, subscriptions: [
       eg.subscription(eg.stream(streamId: 1, name: 'd'), pinToTop: true),
-      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: false),
+      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: true),
       eg.subscription(eg.stream(streamId: 3, name: 'b'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 4, name: 'a'), pinToTop: true),
+    ]);
+    check(isPinnedHeaderInTree()).isTrue();
+
+    final streamListItems = tester.widgetList<SubscriptionItem>(find.byType(SubscriptionItem)).toList();
+    check(streamListItems.map((e) => e.subscription.streamId)).deepEquals([4, 3, 2, 1]);
+  });
+
+  testWidgets('pinned subscriptions are sorted case insensitively', (tester) async {
+    await setupStreamListPage(tester, subscriptions: [
+      eg.subscription(eg.stream(streamId: 1, name: 'D'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 3, name: 'B'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 4, name: 'a'), pinToTop: true),
+    ]);
+    check(isPinnedHeaderInTree()).isTrue();
+
+    final streamListItems = tester.widgetList<SubscriptionItem>(find.byType(SubscriptionItem)).toList();
+    check(streamListItems.map((e) => e.subscription.streamId)).deepEquals([4, 3, 2, 1]);
+  });
+
+  testWidgets('unpinned subscriptions are sorted', (tester) async {
+    await setupStreamListPage(tester, subscriptions: [
+      eg.subscription(eg.stream(streamId: 1, name: 'd'), pinToTop: false),
+      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: false),
+      eg.subscription(eg.stream(streamId: 3, name: 'b'), pinToTop: false),
       eg.subscription(eg.stream(streamId: 4, name: 'a'), pinToTop: false),
     ]);
+    check(isUnpinnedHeaderInTree()).isTrue();
```

Comment:
Also as part of keeping each test focused, there's no need to add this line here. The headers are checked in a different pair of test cases above. (And they're checked more thoroughly there, checking both absence and presence.)